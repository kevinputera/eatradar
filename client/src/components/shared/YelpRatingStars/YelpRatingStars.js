import React from 'react';

import ZERO from './large_0.png';
import ONE from './large_1.png';
import ONE_HALF from './large_1_half.png';
import TWO from './large_2.png';
import TWO_HALF from './large_2_half.png';
import THREE from './large_3.png';
import THREE_HALF from './large_3_half.png';
import FOUR from './large_4.png';
import FOUR_HALF from './large_4_half.png';
import FIVE from './large_5.png';

function getStars(rating) {
  if (!rating || rating < 0.5) {
    return ZERO;
  }
  if (rating < 1.25) {
    return ONE;
  }
  if (rating < 1.75) {
    return ONE_HALF;
  }
  if (rating < 2.25) {
    return TWO;
  }
  if (rating < 2.75) {
    return TWO_HALF;
  }
  if (rating < 3.25) {
    return THREE;
  }
  if (rating < 3.75) {
    return THREE_HALF;
  }
  if (rating < 4.25) {
    return FOUR;
  }
  if (rating < 4.75) {
    return FOUR_HALF;
  }
  return FIVE;
}

function YelpRatingStars(props) {
  const stars = getStars(props.rating);
  return (
    <div className="yelp-rating-stars">
      <img
        src={stars}
        width={props.large ? '91px' : '77px'}
        alt="Yelp star rating"
      />
    </div>
  );
}

export default YelpRatingStars;
