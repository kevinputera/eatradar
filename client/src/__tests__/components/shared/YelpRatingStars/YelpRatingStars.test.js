import React from 'react';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';

import YelpRatingStars from '../../../../components/shared/YelpRatingStars/YelpRatingStars';

import ONE from '../../../../components/shared/YelpRatingStars/large_1.png';

describe('tests for YelpRatingStars component', () => {
  test('render must match snapshot', () => {
    const rating = 3.2;

    const treeLarge = create(<YelpRatingStars large rating={rating} />).toJSON();
    expect(treeLarge).toMatchSnapshot();

    const treeWithRating = create(<YelpRatingStars rating={rating} />).toJSON();
    expect(treeWithRating).toMatchSnapshot();

    const treeWithoutRating = create(<YelpRatingStars />).toJSON();
    expect(treeWithoutRating).toMatchSnapshot();
  });

  test('png used must match rating', () => {
    const rating = 1.1;
    const wrapper = shallow(<YelpRatingStars rating={rating} />);
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(ONE);
  });
});
