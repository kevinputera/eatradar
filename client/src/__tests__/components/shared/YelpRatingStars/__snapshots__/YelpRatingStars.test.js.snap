// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for YelpRatingStars component render must match snapshot 1`] = `
<div
  className="yelp-rating-stars"
>
  <img
    alt="Yelp star rating"
    src="large_3.png"
    width="91px"
  />
</div>
`;

exports[`tests for YelpRatingStars component render must match snapshot 2`] = `
<div
  className="yelp-rating-stars"
>
  <img
    alt="Yelp star rating"
    src="large_3.png"
    width="77px"
  />
</div>
`;

exports[`tests for YelpRatingStars component render must match snapshot 3`] = `
<div
  className="yelp-rating-stars"
>
  <img
    alt="Yelp star rating"
    src="large_0.png"
    width="77px"
  />
</div>
`;
